<!-- todo: will want to use commented out openni_launch section instead of openni_camera to set up the Kinect, but currently the demo seems to ignore it -->

<!-- this launch file is based on http://ros.org/wiki/turtlebot_block_manipulation/Tutorials/TurtleBotBlockManipulationDemo#Making_your_Own_Launch_File -->

<launch>
  <include file="$(find pr2lite_arm_navigation)/launch/pr2lite_planning_environment.launch" />
  <include file="$(find pr2lite_arm_navigation)/launch/move_left_arm.launch" />
  <include file="$(find pr2lite_arm_navigation)/launch/move_right_arm.launch" />

  <!-- Launch the camera drivers (take this out if the camera you're using is *on* the TurtleBot -->
  <include file="$(find openni_launch)/launch/openni.launch">
    <arg name="camera" value="kinect" />
  </include>
<!--
    <arg name="publish_tf" default="false"/>
  <include file="$(find openni_camera)/launch/openni_node.launch" />
-->
  
  <!-- hack (that doesn't work) to fix tf transform -->
<!--
  <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0 0 0 0 kinect_rgb_optical_frame arm_base_link 100" />
  <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0 0 0 0 kinect_rgb_optical_link right_arm_shelf_link 100" />
-->
  
  <!-- Launch all the actions. (originally just launch/block_manipulation_old_kinect.launch) -->
  <node name="block_detection_action_server" pkg="pr2lite_block_manipulation" type="block_detection_action_server" output="screen">
    <remap from="/camera/depth_registered/points" to="/kinect/depth_registered/points" />
<!--
    <remap from="/kinect/depth_registered/points" to="camera/depth_registered/points" />
-->
  </node>

  <include file="$(find pr2lite_block_manipulation)/launch/pick_and_place.launch" />
  
  <node name="interactive_manipulation_action_server" pkg="pr2lite_block_manipulation" type="interactive_manipulation_action_server" output="screen">
  </node>

  <!-- Launch the server that calls all the actions. -->
  <node name="block_manipulation_action_demo" pkg="pr2lite_block_manipulation" type="block_manipulation_action_demo" output="screen">
    <param name="arm_link" value="/right_arm_shelf_link" /> <!-- replacement for /arm_base_link -->
    <param name="table_height" value="-0.53" /> <!-- Height of the table (in m) relative to /arm_base_link -->
    <param name="block_size" value="0.0381" /><!-- Side dimension of the blocks (in m) -->
    <param name="gripper_open" value="0.042" /><!-- Width (in m) to open the gripper to before grasping -->
    <param name="gripper_closed" value="0.024" /><!-- Width (in m) to close the gripper to while grasping -->
    <param name="z_up" value="0.12" /><!-- Height to raise the arm to (in m) relative to /arm_base_link -->
  </node>
  
</launch>
